website/source/guides/tls.html.md

```diff
@@ -0,0 +1,425 @@
+
+```text
+Error querying node status: Get http://127.0.0.1:4646/v1/nodes: malformed HTTP response "\x15\x03\x01\x00\x02\x02"
+```
+
+This is because the Nomad CLI defaults to communicating via HTTP instead of
+HTTPS. We can configure the local Nomad client to connect using TLS and specify
+our custom keys and certificates using the command line:
+
+```shell
+$ nomad node-status -ca-cert=nomad-ca.pem -client-cert=cli.pem -client-key=cli-key.pem -addr=https://127.0.0.1:4646
+```
+
+This process can be cumbersome to type each time, so the Nomad CLI also
+searches environment variables for default values. Set the following
+environment variables in your shell:
+
+```shell
+$ export NOMAD_ADDR=https://localhost:4646
+$ export NOMAD_CACERT=nomad-ca.pem
+$ export NOMAD_CLIENT_CERT=cli.pem
+$ export NOMAD_CLIENT_KEY=cli-key.pem
+```
+
+* `NOMAD_ADDR` is the URL of the Nomad agent and sets the default for `-addr`.
+* `NOMAD_CACERT` is the location of your CA certificate and sets the default
+  for `-ca-cert`.
+* `NOMAD_CLIENT_CERT` is the location of your CLI certificate and sets the
+  default for `-client-cert`.
+* `NOMAD_CLIENT_KEY` is the location of your CLI key and sets the default for
+  `-client-key`.
+
+After these environment variables are correctly configured, the CLI will
+respond as expected:
+
+```text
+$ nomad node-status
+ID        DC   Name   Class   Drain  Status
+237cd4c5  dc1  nomad  <none>  false  ready
+
+$ nomad init
+Example job file written to example.nomad
+vagrant@nomad:~$ nomad run example.nomad
+==> Monitoring evaluation "e9970e1d"
+    Evaluation triggered by job "example"
+    Allocation "a1f6c3e7" created: node "237cd4c5", group "cache"
+    Evaluation within deployment: "080460ce"
+    Evaluation status changed: "pending" -> "complete"
+==> Evaluation "e9970e1d" finished with status "complete"
+```
+
+## Server Gossip
+
+At this point all of Nomad's RPC and HTTP communication is secured with mTLS.
+However, Nomad servers also communicate with a gossip protocol, Serf, that does
+not use TLS:
+
+* HTTP - Used to communicate between CLI and Nomad agents. Secured by mTLS.
+* RPC - Used to communicate between Nomad agents. Secured by mTLS.
+* Serf - Used to communicate between Nomad servers. Secured by a shared key.
+
+Nomad server's gossip protocol use a shared key instead of TLS for encryption.
+This encryption key must be added to every server's configuration using the
+[`encrypt`](/docs/agent/configuration/server.html#encrypt) parameter or with
+the [`-encrypt` command line option](/docs/commands/agent.html).
+
+The Nomad CLI includes a `keygen` command for generating a new secure gossip
+encryption key:
+
+```text
+$ nomad keygen
+cg8StVXbQJ0gPvMd9o7yrg==
+```
+
+Alternatively, you can use any method that base64 encodes 16 random bytes:
+
+```text
+$ dd if=/dev/urandom bs=16 count=1 status=none | base64
+LsuYyj93KVfT3pAJPMMCgA==
+$ python -c 'import base64; print base64.b64encode(open("/dev/urandom").read(16))'
```

Comment:
I'm with @jvoorhis. I added the openssl example. I think people will get the point that base64'ing 16 bytes of entropy can be accomplished using just about anything.